umm-spec-lib/src/cmr/umm_spec/xml_to_umm_mappings/iso_shared/doi.clj

```diff
@@ -0,0 +1,40 @@
+(ns cmr.umm-spec.xml-to-umm-mappings.iso-shared.doi
+  "Functions for parsing UMM DOI records out of ISO 19115 and ISO SMAP XML documents."
+  (:require
+   [cmr.common.util :as util]
+   [cmr.common.xml.simple-xpath :refer [select]]
+   [cmr.common.xml.parse :refer [value-of]]))
+
+(def doi-namespace
+  "DOI namespace."
+  "gov.nasa.esdis.umm.doi")
+
+(defn- is-doi-field?
+  "Returns true if the given gmd-id is for a DOI field."
+  [gmd-id]
+  (and (= (value-of gmd-id "gmd:MD_Identifier/gmd:description/gco:CharacterString") "DOI")
+       (= (value-of gmd-id "gmd:MD_Identifier/gmd:codeSpace/gco:CharacterString") doi-namespace)))
+
+(defn parse-doi
+  "There could be multiple CI_Citations. Each CI_Citation could contain multiple gmd:identifiers.
```

Comment:
Super helpful docstring 👍 

Might be worth adding a sentence about why we're picking the first DOI for now

Reply:
I can't take credit for this docstring. I don't think there's anything special about the first, that's just the pattern used when there are multiple of something in one format, but we can only use one in another format.